src/display/annotation_layer.js

```diff
@@ -544,40 +544,42 @@ class WidgetAnnotationElement extends AnnotationElement {
   }
 
   _setEventListener(element, baseName, eventName, valueGetter) {
-    if (this.data.actions && eventName.replace(" ", "") in this.data.actions) {
-      if (baseName.includes("mouse")) {
-        // Mouse events
-        element.addEventListener(baseName, event => {
-          window.dispatchEvent(
-            new CustomEvent("dispatchEventInSandbox", {
-              detail: {
-                id: this.data.id,
-                name: eventName,
-                value: valueGetter(event),
-                shift: event.shiftKey,
-                modifier: this._getKeyModifier(event),
-              },
-            })
-          );
+    if (this.data.actions[eventName.replace(" ", "")] === undefined) {
+      return;
+    }
+    if (baseName.includes("mouse")) {
+      // Mouse events
+      element.addEventListener(baseName, event => {
+        this.linkService.eventBus?.dispatch("dispatcheventinsandbox", {
```

Comment:
I think we should have at least a warning in the console if eventBus is null.
Or if we expect that eventBus is neither null nor undefined, then you can remove the `?` and we'll get an exception.

Reply:
That's not really a problem though, since the `AnnotationLayer` *must* be initialized with either a `PDFLinkService` or a `SimpleLinkService` instance (the latter having only dummy-methods).
Note how the existing code simply assumes that *either* is available, and the optional chaining added here (and elsewhere) is simply intended to prevent errors in the `SimpleLinkService`-case.